DuckDuckGo/Bookmarks/Model/BookmarkList.swift

```diff
@@ -149,6 +157,15 @@ struct BookmarkList {
         return allBookmarkURLsOrdered
     }
 
+    private mutating func updateLowercasedItemsDict() {
```

Comment:
Rather than going over the full dictionary from scratch, when only a single itemDict entry is modified, we could perform a simpler operation. Could you add separate methods that handle single items? Something like this, or similar:

```swift
func addToLowercaseItemsDict(_ bookmark: Bookmark)
func removeFromLowercaseItemsDict(_ bookmark: Bookmark)
func updateLowercaseItemsDict(_ bookmark: Bookmark, oldURL: URL)
```

Reply:
I don't recommend this approach as it may introduce diverge data. Going over the full dictionary happens only when there is a bookmark change, which isn't that often

Reply:
Then perhaps it's worth to keep bookmarks index as a separate class and handle updates there. Since that would be out of scope for this task, I'd be happy to go with what we have now, since your solution fixes the issue which is the most important thing here.